src/FileWriter.php

```diff
@@ -25,21 +25,25 @@ final class FileWriter
      */
     public static function writeFile($file, $content, $chmod = 0666)
     {
-        $tmp = tempnam(sys_get_temp_dir(), md5($file));
+        $dir = dirname($file);
+        $tmp = tempnam($dir, $file);
```

Comment:
I'd rather make it static (probably, up to 3 letters, see the Windows limitation in the [documentation](refsect1-function.tempnam-parameters)). Ideally, the prefix should be derived from the application name so that the server admin knows what these files belong to. Since the current API doesn't allow setting the prefix, it could be derived from the library name.

Reply:
@morozov not sure if I get your point here. 
We need random file names, to make sure we are not writing to the same file from two different processes. We want to avoid collisions and blocking and broken files on writing.
Can you provide an example how you see it? Thanks!

Reply:
It should be `$tmp = tempnam($dir, 'foo');` where `foo` is the name of the app or the library. See https://3v4l.org/eJLnZ.

Reply:
That's why I've used `md5` on file before, as the file can be really anything, but it additional operation and 32 chars. Should I go then for now with `wsw` (from webimpress/safe-writer)?

Reply:
> Should I go then for now with `wsw` (from webimpress/safe-writer)?

Yes, this is exactly what I had in mind.